### ISSUE TITLE
WaitForUpdates stuck for more than 2 days

### ISSUE THREAD

**Issue by rahulrane50** [vmware/pyvmomi#548]
[pyvmomi version = 6.0.0.2016.6]
I'm facing one issue while creating snapshot. The api got stuck for more than 2 days or in some cases until i kill the process. I did some analysis here and need your opinion on that. Let's start with thread dump : 
File: "pyVmomi/VmomiSupport.py", line 560, in __call__
File: "pyVmomi/VmomiSupport.py", line 383, in _InvokeAccessor
File: "pyVmomi/StubAdapterAccessorImpl.py", line 42, in InvokeAccessor
File: "pyVmomi/VmomiSupport.py", line 566, in <lambda>
File: "pyVmomi/VmomiSupport.py", line 375, in _InvokeMethod
File: "pyVmomi/SoapAdapter.py", line 1282, in InvokeMethod
File: "pyVmomi/SoapAdapter.py", line 783, in Deserialize
File: "pyVmomi/SoapAdapter.py", line 456, in ReadDocument
File: "httplib.py", line 522, in read
File: "httplib.py", line 569, in _read_chunked
File: "httplib.py", line 619, in _safe_read
File: "socket.py", line 377, in read
File: "ssl.py", line 215, in recv
File: "ssl.py", line 136, in read

Looking at dump it looks like socket recv got stuck. After digging deep i found that there is no timeout value set on this socket. The class HTTPSConnectionWrapper in SoapAdapter sets timeout value but only if we pass any ssl params. In my case i don't have any extra ssl paramters to pass to pyVim Connect call. 
I also looked at connection_pool_timeout but after looking at pyVim Connect call there is provision to set this parameter which is making it set to default 900 secs. I am assuming here even 900 secs is set that is not set on the socket created by httplib hence my WaitForUpdates call stuck. After looking into httplib code if timeout value is not passed then not even global timeout is set.

I also explored WaitForUpdatesEx api. But here I'm not sure if setting maxWaitSeconds would set timeout for the httplib socket. 

My suggestion here would be either provide explicit "timeout" parameter to pyVim Connect/SmartConnect api and pass it to HTTPSConnectionWrapper or set some default timeout value to this socket.

Please correct me if I'm missing here something. I would even like to contribute this thing if given a chance.

**Comment by tianhao64** [comment]
@rahulrane50 You are right. It seems SoapStubAdapter should take a timeout parameter and pass it to the socket level. It'd be really helpful if you could create a PR to fix it. Btw, HTTPSConnectionWrapper is removed in 6.5 release. So I recommend you to upgrade to the latest release. It should work for older vSphere too.
For the time being, you can pass maxWaitSeconds to WaitForUpdatesEx to avoid getting blocked forever when there is no update occur in the environment.

**Comment by rahulrane50** [comment]
@tianhao64 I tried pyvmomi 6.5 as well but there also i could not find socket timeout. Also tired with WaitForUpdatesEx api with maxWaitSeconds but I assume that will not solve my problem sine task is in progress but socket.recv has stuck.
